examples/html/src/index.ts

```diff
@@ -0,0 +1,62 @@
+import { SubspaceClient, Identity } from "subspace.js";
+
+const NODE_WS_PROVIDER = 'ws://localhost:9944'
+const FARMER_WS_PROVIDER = 'ws://localhost:9944'
+
+const htmlObjectId = document.getElementById('objectId') as HTMLHeadElement
+const inputImageHtml = document.getElementById('inputImage') as HTMLImageElement
+const htmlResultImage = document.getElementById('resultImage') as HTMLImageElement
+
+(document.getElementById("fileInput") as HTMLImageElement).addEventListener("change", loadFileImage, false);
+(document.getElementById("putObjectBtn") as HTMLButtonElement).addEventListener("click", putObject, false);
+(document.getElementById("getObjectBtn") as HTMLButtonElement).addEventListener("click", getObject, false);
+
+let subspaceClient: SubspaceClient
+let imageData: any, object: any, objectId: any
+
+async function putObject() {
+    if (!imageData) return alert('Select an image first.')
+    objectId = await subspaceClient.putObject(imageData)
+    htmlObjectId.innerHTML = 'objectId: ' + objectId
+}
+
+async function getObject() {
+    if (!objectId) return alert('Put the file first.')
+    object = await subspaceClient.getObject(objectId)
+
+    const b64encoded = btoa(String.fromCharCode.apply(null, object))
+    htmlResultImage.src = `data:image/*;base64,${b64encoded}`
+}
+
+function loadFileImage(e: any) {
+    let files = e.target.files
+    if (files && files.length) {
+        const image = files[0]
+        let frAsUrl = new FileReader()
+        let frAsArrayBuffer = new FileReader()
+
+        frAsUrl.onload = function () {
+            inputImageHtml.src = frAsUrl.result?.toString() || ''
+        }
+        frAsUrl.readAsDataURL(image)
+
+        frAsArrayBuffer.onload = function () {
+            if (frAsArrayBuffer.result)
+                imageData = new Uint8Array(frAsArrayBuffer.result as ArrayBuffer)
+        }
+        frAsArrayBuffer.readAsArrayBuffer(image)
+    }
+}
+
+async function init() {
+    const identity = await Identity.fromWeb3()
+    subspaceClient = await SubspaceClient.connect(
+        identity,
+        NODE_WS_PROVIDER,
+        FARMER_WS_PROVIDER,
+    )
+    const address = identity.getKeyringPair().address;
+    (document.getElementById('injectedAccount') as HTMLHeadingElement).innerHTML = address
+}
+
+setTimeout(init, 2000)
```

Comment:
Is this timeout to make sure crypto wasm in polkadot.js is loaded? I'm almost certain it will fail randomly, you need to wait for it to load using corresponding API: https://github.com/subspace/subspace-relayer/blob/2fbf38eec744acf12933436dcb59924b4ee4b371/backend/src/poolSigner/signingWorker.ts#L19

Reply:
Is for `polkadot/extension-dapp`. looks like the dynamic import for the extension is not working correctly, the timeout was fot that

https://github.com/subspace/subspace.js/blob/641a6d74be31385feeb5a6245389de7a5e343d21/src/identity.ts#L26-L31

Reply:
This needs to be implemented properly. Having random timeouts with not explanation whatsoever is a bad practice and not great for an example app.

Reply:
I have removed the timeout. 
The problem here is a difference of milliseconds between the `injection` of the extension and the `web3Enabled` method checking the injection status, it was returning _zero extensions detected and zero accounts_, failing to load the library from the `Identity.fromWeb3 `method. 

It turns out that the ECMAScript build of `extension-dapp` does not have this problem. After reviewing, I found a way to inject an extension manually using the `extension-inject` package, but had to provide exact **name and version**, it did not seem like a good idea as for each new version of the extension, you would have to update the library.

I have found an _optional_ parameter in the `web3Enabled` call that solves the problem. I explain it in the following comment. It works and is just 1 extra line of code. 

I will check if there is a better way to solve this, maybe from the lib itself.